internal/provider/serverless_cluster_resource.go

```diff
@@ -0,0 +1,730 @@
+							},
+							"disabled": schema.BoolAttribute{
+								MarkdownDescription: "Whether the public endpoint is disabled.",
+								Optional:            true,
+							},
+						},
+					},
+					"private_endpoint": schema.SingleNestedAttribute{
+						MarkdownDescription: "The private endpoint for connecting to the cluster.",
+						Computed:            true,
+						PlanModifiers: []planmodifier.Object{
+							objectplanmodifier.UseStateForUnknown(),
+						},
+						Attributes: map[string]schema.Attribute{
+							"host": schema.StringAttribute{
+								MarkdownDescription: "The host of the private endpoint.",
+								Computed:            true,
+							},
+							"port": schema.Int64Attribute{
+								MarkdownDescription: "The port of the private endpoint.",
+								Computed:            true,
+							},
+							"aws_endpoint": schema.SingleNestedAttribute{
+								MarkdownDescription: "Message for AWS PrivateLink information.",
+								Computed:            true,
+								Attributes: map[string]schema.Attribute{
+									"service_name": schema.StringAttribute{
+										MarkdownDescription: "The AWS service name for private access.",
+										Computed:            true,
+									},
+									"availability_zone": schema.ListAttribute{
+										MarkdownDescription: "The availability zones that the service is available in.",
+										Computed:            true,
+										ElementType:         types.StringType,
+									},
+								},
+							},
+							"gcp_endpoint": schema.SingleNestedAttribute{
+								MarkdownDescription: "Message for GCP PrivateLink information.",
+								Computed:            true,
+								Attributes: map[string]schema.Attribute{
+									"service_attachment_name": schema.StringAttribute{
+										MarkdownDescription: "The target GCP service attachment name for private access.",
+										Computed:            true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"root_password": schema.StringAttribute{
+				MarkdownDescription: "The root password of the cluster.",
+				Optional:            true,
+				Sensitive:           true,
+			},
+			"encryption_config": schema.SingleNestedAttribute{
+				MarkdownDescription: "The encryption settings for the cluster.",
+				Optional:            true,
+				Computed:            true,
+				PlanModifiers: []planmodifier.Object{
+					objectplanmodifier.UseStateForUnknown(),
+				},
+				Attributes: map[string]schema.Attribute{
+					"enhanced_encryption_enabled": schema.BoolAttribute{
+						MarkdownDescription: "Whether enhanced encryption is enabled.",
+						Optional:            true,
+						Computed:            true,
+					},
+				},
+			},
+			"high_availability_type": schema.StringAttribute{
+				MarkdownDescription: "The high availability type of the clusterV1beta1. ZONAL: High availability within a single zone. REGIONAL: High availability across multiple zones within a region",
+				Required:            true,
```

Comment:
it is not required i think.

Reply:
I tested it, and it works without filling in the field, but swagger doesn't mark it as optional.